src/caffe/layers/batch_norm_layer.cu

```diff
@@ -85,8 +85,10 @@ void BatchNormLayer<Dtype>::Forward_gpu(const vector<Blob<Dtype>*>& bottom,
   caffe_gpu_div(temp_.count(), top_data, temp_.gpu_data(), top_data);
   // TODO(cdoersch): The caching is only needed because later in-place layers
   //                 might clobber the data.  Can we skip this if they won't?
-  caffe_copy(x_norm_.count(), top_data,
+  if (!use_global_stats_) {
```

Comment:
@daijifeng001 Can you explain the reason for doing this? Apparently, the test `BatchNormLayerTest/3.TestForwardInplace` fails because of this.